WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Helpers/BlogDashboardState.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+class BlogDashboardState {
```

Comment:
I feel this class is a bit confusing. To avoid it breaking in the future I suggest adding unit tests just for it.

Reply:
Also I can see that everywhere we use `BlogDashboardState.shared`, we already have a reference to `Blog`.
Instead of singleton, what about adding this as a property inside `Blog`? 
Meaning each `Blog` would have an instance of `BlogDashboardState` this way we wouldn't have to reset it with each site switch. WDYT?

Reply:
> I feel this class is a bit confusing. To avoid it breaking in the future I suggest adding unit tests just for it.

Will do!

> what about adding this as a property inside Blog?

I'll explore that. My idea would be to just use an in-memory `dashboard` property though, instead of adding properties to this Entity and persisting them on Core Data. Wdyt?

Reply:
> in-memory dashboard property

Yes, that's what I had in mind as well 👌

Reply:
@hassaanelgarem I added the "dashboard state" as a blog property and added unit tests for it.

It makes 100% sense to tie the state to the blog, I can see that this was probably a big part of your confusion. You can follow the commits on this PR after your last comment. Let me know what do you think, thanks!

Reply:
I think it's much clearer now 👍